src/plugins/discourse/mirror.test.js

```diff
@@ -568,4 +654,548 @@ describe("plugins/discourse/mirror", () => {
+      expect(repo.topics()).toEqual([topic4]);
+    });
+
+    it("mirrors category definition posts and normal posts from the fetcher", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+      const p1 = fetcher.addCategory(4);
+      const p2 = fetcher.addCategory(5);
+      const p3 = fetcher.addPost(5, 1);
+      const p4 = fetcher.addPost(10, null);
+      const p5 = fetcher.addPost(11, null);
+      const p6 = fetcher.addPost(11, 1);
+      await mirror.update(reporter);
+      const posts = [
+        fetcher._post(p1),
+        fetcher._post(p2),
+        fetcher._post(p3),
+        fetcher._post(p4),
+        fetcher._post(p5),
+        fetcher._post(p6),
+      ];
+      expect(repo.posts()).toEqual(posts);
+    });
+
+    it("mirrors updates normal posts, not category definition posts", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+
+      const p1 = fetcher.addCategory(4);
+      const p2 = fetcher.addCategory(5);
+      const p4 = fetcher.addPost(10, null);
+      const p5 = fetcher.addPost(11, null);
+      await mirror.update(reporter);
+
+      // Category definition post we don't expect to be updated.
+      fetcher.addPost(5, 1);
+      const p6 = fetcher.addPost(11, 1);
+      await mirror.update(reporter);
+
+      const posts = [
+        fetcher._post(p1),
+        fetcher._post(p2),
+        fetcher._post(p4),
+        fetcher._post(p5),
+        fetcher._post(p6),
+      ];
+      expect(repo.posts()).toEqual(posts);
+    });
+
+    it("mirrors updates normal post edits, not category definition post edits", async () => {
+      const {mirror, fetcher, reporter, repo} = example();
+
+      const p1 = fetcher.addCategory(4);
+      const p2 = fetcher.addCategory(5);
+      const p3 = fetcher.addPost(10, null);
+      const p4 = fetcher.addPost(11, null);
+      const expectedBefore = [
+        fetcher._post(p1),
+        fetcher._post(p2),
+        fetcher._post(p3),
+        fetcher._post(p4),
+      ];
+      await mirror.update(reporter);
+      const actualBefore = repo.posts();
+
+      // Category definition post we don't expect to be updated.
+      fetcher.editPost(p2, {
+        cooked: `<p>Edit: changed category definition post.</p>`,
+        timestampMs: Date.now(),
+      });
+      fetcher.editPost(p4, {
+        cooked: `<p>Edit: changed normal post.</p>`,
+        timestampMs: Date.now(),
+      });
+      const expectedAfter = [
+        fetcher._post(p1),
+        expectedBefore[1],
+        fetcher._post(p3),
+        fetcher._post(p4),
+      ];
+      await mirror.update(reporter);
+      const actualAfter = repo.posts();
+
+      expect(actualBefore).toEqual(expectedBefore);
+      expect(actualAfter).toEqual(expectedAfter);
+      // Sanity check, make sure the category post was edited in the mock fetcher.
+      expect(fetcher._post(p2)).not.toEqual(expectedAfter[1]);
+    });
+
+    it("provides usernames for all active users", async () => {
```

Comment:
what would happen if a user only posted in a category definition topic?

Reply:
Although the process of _discovering_ category definition topics is different. The underlying structure is still a topic with posts which have authors. So these users will be included. Keeping in mind the (re)load interval of category definition topics.